### PR TITLE
(MODULES-3430) Do not set default location to resource title

### PR DESCRIPTION
**Pull request by ferventcoder**
When ensuring a chocolateysource is present, location is a required value
and should be explicitly set. Previously, location would just take on the
default value of the resource title. This could be alarming to some users
who may set a resource like:

~~~puppet
chocolateysource {'chocolatey':}
~~~

And be later surprised to find that it replaces the location value with
`chocolatey` instead of whatever it was originally, in this case that
value was `https://chocolatey.org/api/v2/`. Due to this, which could be
both alarming and upsetting, the resource should instead explicitly
require a location when `ensure => present`. When disabling or
removing a chocolateysource, it should be fine to simply name the
source to disable/remove.

### CODE REVIEW

**Comment by ferventcoder** [comment]
@puppetlabs/windows take a look.

**Comment by ferventcoder** [comment]
And now we are good.

File: lib/puppet/type/chocolateysource.rb [hunk 1]
```diff
@@ -120,6 +111,10 @@ def insync?(is)
     if provider.respond_to?(:validate)
       provider.validate
     end
+
+    if (self[:ensure].to_sym == :present && (self[:location].nil? || self[:location] == ''))
+      raise ArgumentError, "A non-empty location must be specified when not disabling/removing a source."
```
Comment (glennsarti) [inline lib/puppet/type/chocolateysource.rb]: This error message is very confusing as it includes a double negative. Would prefer;
`A location must be specified when setting a source`

File: lib/puppet/type/chocolateysource.rb [hunk 2]
```diff
@@ -120,6 +111,10 @@ def insync?(is)
     if provider.respond_to?(:validate)
       provider.validate
     end
+
+    if (self[:ensure].to_sym == :present && (self[:location].nil? || self[:location] == ''))
```
Comment (glennsarti) [inline lib/puppet/type/chocolateysource.rb]: Do we care about a location of whitespace? e.g. `location => ' ',`

Comment (ferventcoder) [inline lib/puppet/type/chocolateysource.rb]: Does it error properly or does it let it through?

Comment (glennsarti) [inline lib/puppet/type/chocolateysource.rb]: Let's it through and set's it in the choco config.

File: lib/puppet/type/chocolateysource.rb (hunk 1 above)
Comment (ferventcoder) [inline lib/puppet/type/chocolateysource.rb]: Not exactly a double negative. 

non-empty is an adjective describing location.

The second part is about when it must be specified, as in when it is ensure => present (or the default of that when it is not specified). It was becoming a lot of words to express the same thing more succinctly.

File: lib/puppet/type/chocolateysource.rb (hunk 2 above)
Comment (ferventcoder) [inline lib/puppet/type/chocolateysource.rb]: I can update this.

**Pull request merged**